pkg/query/union.go

```diff
@@ -23,7 +23,8 @@ func (u *Union) addSubIterator(subIt Iterator) {
 }
 
 func (u *Union) CheckImpl(ctx *Context, resourceIDs []string, subjectID string) (RelationSeq, error) {
-	remaining := resourceIDs
+	remaining := make([]string, len(resourceIDs))
+	copy(remaining, resourceIDs)
```

Comment:
Why is this necessary? Are we mutating this somewhere?

Reply:
Yep, right here with slicez.Delete; which causes a bug because it could be used again. Could also change it to create a new incremental list, up to you for what you prefer.